### PR TITLE
Add map layer action for current GPS feature destination layer for "Update GPS Information"

### PR DESCRIPTION
**Pull request by nyalldawson**
Triggering this action from the attribute dialog for a newly
created feature in the GPS destination layer causes the geometry
of that feature to be refreshed with the CURRENT GPS location.

Designed for situations where populating the form attributes
can be a lengthy process and in the meantime the GPS receiver
has moved, and a user needs to re-sync the location of the
new feature back to the GPS location before saving.

This action ONLY shows up when creating new features in the GPS destination layer, so as not to clutter the UI in other cirucmstances.

Sponsored by NIWA

Also includes an API extension for the QgsMapLayerAction classes so that an execution context can be passed to actions.

### CODE REVIEW

**Comment by nyalldawson** [comment]
Now includes even **more** cleanups to the app GPS logging classes!

**Comment by bettellam** [comment]
Ciao @nyalldawson, you could also use system variables where the values of last GPS position are "saved" and with these values "populate" fields of current layer.
Qfield does this with the @position_XXXXX variables.

**Comment by nyalldawson** [comment]
@bettellam GPS related variables would be awesome, I agree! There's still a considerable amount of work required before they could be added in QGIS desktop though, and ultimately address a slightly different use case to what this PR is intended to address.

**Comment by bettellam** [comment]
@nyalldawson, I am following with curiosity and admiration all work you are doing with the new GPS Toolbar (and not only that).
Lots of PR, lots of details to fix to make this new tool easy for end users to use! A great job!
Thank you
Grazie, ciao

File: src/app/gps/qgsappgpsdigitizing.cpp [hunk 1]
```diff
@@ -26,14 +26,66 @@
...
+}
+
+bool QgsUpdateGpsDetailsAction::canRunUsingLayer( QgsMapLayer * ) const
+{
+  return false;
+}
+
+bool QgsUpdateGpsDetailsAction::canRunUsingLayer( QgsMapLayer *layer, const QgsMapLayerActionContext &context ) const
+{
+  return mConnection && mConnection->isConnected() && context.attributeDialog() && context.attributeDialog()->attributeForm()->mode() == QgsAttributeEditorContext::Mode::AddFeatureMode
```
[the remark this answers is not in the document]
Comment (nyalldawson) [reply src/app/gps/qgsappgpsdigitizing.cpp:52]: Actually it works ok for those geometry types too, assuming that there's a current GPS track with 2 or 3 vertices respectively. If not, then the message bar in the dialog is used to show a warning explaining this requirement.

**Review by nirvn — APPROVED** [review]

**Pull request merged**